AK/FloatingPointStringConversions.cpp

```diff
@@ -1543,123 +1529,170 @@ class MinimalBigInt {
 
     void multiply_with_power_of_5(u32 exponent)
     {
-        // FIXME: We might be able to store a bigger power of 5 but this would
-        //        require a wide multiply, so perhaps using u4096 would be
-        //        better to get wide multiply and not duplicate logic.
-        static constexpr Array<u64, 28> power_of_5 = {
-            1ul,
-            5ul,
-            25ul,
-            125ul,
-            625ul,
-            3125ul,
-            15625ul,
-            78125ul,
-            390625ul,
-            1953125ul,
-            9765625ul,
-            48828125ul,
-            244140625ul,
-            1220703125ul,
-            6103515625ul,
-            30517578125ul,
-            152587890625ul,
-            762939453125ul,
-            3814697265625ul,
-            19073486328125ul,
-            95367431640625ul,
-            476837158203125ul,
-            2384185791015625ul,
-            11920928955078125ul,
-            59604644775390625ul,
-            298023223876953125ul,
-            1490116119384765625ul,
-            7450580596923828125ul,
-        };
+        // FIXME: Using UFixedBigInt here does not feel right. We need to nicely (without
+        //        reintrepret_cast and ifs) convert u64 to NativeWord (if NativeWord == u32)
```

Comment:
is `NativeWord == u32` even possible by now?

Reply:
shouldnʼt we target wasm64 only if the system is meant to be 64-bit only? (at least at the moment being). 🤔

Reply:
The main problem is that there is no support for `__int128` on wasm, so although wasm is effectively 64-bit (to my knowledge), we can't really use the wideness fully.

Reply:
i must confess i did not caught exactly how it relates to `u32` as `NativeWord`… 😔

still: those were all the questions and remarks from me; thanks! 🙇🏻‍♂️

Reply:
https://github.com/SerenityOS/serenity/blob/2258fc273c9eb0db404a8020b8621fab31d1f7f7/AK/BigIntBase.h#L19-L27
To set NativeWord to u64 we need 64-bit platform and __int128 support

Reply:
Since #17330 is already merged, I will need to create a separate commit just to add one comment...

Reply:
no problem with me here; consider that resolved then! 😅